code/datums/diseases/anxiety.dm

```diff
@@ -36,6 +36,15 @@
 			if(prob(2))
 				affected_mob.visible_message("<span class='danger'>[affected_mob] coughs up butterflies!</span>", \
 													"<span class='userdanger'>You cough up butterflies!</span>")
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
-				new /mob/living/simple_animal/butterfly(affected_mob.loc)
+				for(var/i in 1 to 2)
+					var/mob/living/simple_animal/butterfly/B = new(affected_mob.loc)
+					addtimer(CALLBACK(B, /mob/living/simple_animal/butterfly/.proc/decompose), rand(5, 25) SECONDS)
 	return
```

Comment:
While you're here remove this useless return

Reply:
Done!